python/pyspark/sql/tests/test_dataframe.py

```diff
@@ -531,7 +531,7 @@ def test_with_columns(self):
 
         # Type check
         self.assertRaises(TypeError, self.df.withColumns, ["key"])
-        self.assertRaises(AssertionError, self.df.withColumns)
+        self.assertRaises(Exception, self.df.withColumns)
```

Comment:
This wasn't actually testing Spark Connect (because of `self.df` created from the regular Spark session).